solidity/dashboard/src/css/timeline.less

```diff
@@ -0,0 +1,101 @@
+@margin-bottom-between-elements: 1rem;
+@left-border-width: 8px;
+
+.timeline {
+  position: relative;
+
+  &__element {
+    position: relative;
+    min-width: 200px;
+    padding: 1.5rem 2rem;
+    border: 1px solid #AE96E1; // @color-brand-violet-40
+    border-left: @left-border-width solid #633EB1; // @color-brand-violet-80
+    border-radius: 0.5rem;
+    &:extend(.box-shadow);
+    background-color: @white;
+    margin-bottom: @margin-bottom-between-elements;
+    
+
+    &--left {
+      margin-left: 3rem;
+    }
+
+    &--right {
+      // TODO add support for displaying timeline events on the right side and
+      // alternate.
+      margin-left: 3rem;
+    }
+
+    // Drawing the dashed line for the first element. NOTE: only valid when timeline
+    // breeakpoints are centered relative to the event content.
+    &:nth-of-type(1) {
+        &::before {
+          height: unset;
+          top: 50%;
+          bottom: 0;
+          .draw-dashed-vertical-line(@color-grey-30);
+        }
+    }
+
+    // Drawing the dashed line for the last element. NOTE: only valid when timeline
+    // breeakpoints are centered relative to the event content.
+    &:nth-last-of-type(1) {
+      &::before {
+        top: 0;
+        bottom: 50%;
+        height: unset;
+        .draw-dashed-vertical-line(@color-grey-30);
+      }
+    }
+
+    // Drawing the dashed line for the rest elements between the first and last
+    // element. NOTE: only valid when timeline breeakpoints are centered relative to
+    // the event content.
+    &::before {
+      .draw-dashed-vertical-line(@color-grey-30);
+      content: '';
+      position: absolute;
+      width: 2px;
+      left: calc(-3rem - 8px);
+      height: calc(100% + 2rem);
+      top: -1rem;
+      bottom: -1rem;
+      overflow: visible;
+    }
+  }
```

Comment:
The dashed line is not looking good in certain cases (tested only on Chrome).

Examples:
<details>
  <summary>For 1 element:</summary>

![image](https://user-images.githubusercontent.com/40306834/126496571-19424fa3-640d-4d4b-bb4e-92503104dcaf.png)
</details>

<details>
  <summary>For 5 element:</summary>

![image](https://user-images.githubusercontent.com/40306834/126496498-ec44a504-24f2-4b56-bd41-d48623039cf3.png)
</details>

What i propose is to write the line for each element only below it (instead of 50% on top and 50% on bottom). I refactor the code here and there and i ended up with something like this:

```
    // Drawing the dashed line for the first element. NOTE: only valid when timeline
    // breeakpoints are centered relative to the event content.
    &:nth-of-type(1) {
        &::before {
          top: 50%;
          bottom: 0;
          height: calc(100% + 2rem);
          .draw-dashed-vertical-line(@color-grey-30);
        }
    }

    // Drawing the dashed line for the last element. NOTE: only valid when timeline
    // breeakpoints are centered relative to the event content.
    &:nth-last-of-type(1) {
      &::before {
        top: 0;
        bottom: 50%;
        height: unset;
        background: transparent;
      }
    }

    // Drawing the dashed line for the rest elements between the first and last
    // element. NOTE: only valid when timeline breeakpoints are centered relative to
    // the event content.
    &::before {
      .draw-dashed-vertical-line(@color-grey-30);
      content: '';
      position: absolute;
      width: 2px;
      left: calc(-3rem - 8px);
      height: calc(100% + 2rem);
      top: 50%;
      bottom: 0;
      overflow: visible;
    }
  }
```

It makes the lines look ok in both cases that i mentioned.

Reply:
> to write the line for each element only below it (instead of 50% on top and 50% on bottom)

This is only true when all elements have the same height. We need to draw a line with the height of the element so I think the only way is to draw 50% on bottom and 50% on top. I'll try to make the dashed line looks better.

Reply:
For the dashed line element, it looks like it's missing the rounded radius here, like this: 

<img width="165" alt="Screen Shot 2021-07-22 at 2 27 47 PM" src="https://user-images.githubusercontent.com/57226633/126690252-ae059172-26a9-4398-8478-6d797dd03d00.png">

I'm not sure the exact radius is though since Figma does dashed lines like this:
<img width="336" alt="Screen Shot 2021-07-22 at 2 29 02 PM" src="https://user-images.githubusercontent.com/57226633/126690435-0ef75d6d-5d2e-4c13-a289-5185c0b8b057.png">
And doesn't give a radius setting for individual dash marks.

Reply:
Fixed in 3ee96a6.

For one timeline element.
![obraz](https://user-images.githubusercontent.com/57687279/127006963-d445d441-8a63-4c82-ba0d-84129d1b7074.png)

Reply:
> For the dashed line element, it looks like it's missing the rounded radius here, like this:

Regarding rounded dots. Firefox correctly draws a dotted line - with rounded corners. Chrome has a problem with this and it draws a dashed line. What the dotted line looks like depends on the browser and I'm afraid there is nothing we can do with this.

Chrome:
<img width="652" alt="obraz" src="https://user-images.githubusercontent.com/57687279/127008705-2c0facc3-aced-4ae6-8479-60ef23f741a8.png">

Firefox:
![obraz](https://user-images.githubusercontent.com/57687279/127008952-6e10fc9a-7ad7-412e-b280-f9f86ae9ab0f.png)

cc @liz-shinn

Reply:
Ahhh ok, woof! Thx for explaining that quirk, alas cursed chrome